src/test/java/org/junitpioneer/jupiter/resource/ResourcesTests.java

```diff
@@ -0,0 +1,543 @@
+
+		@Override
+		public Resource<Object> create(List<String> arguments) {
+			return new ThrowOnRCloseResource();
+		}
+
+	}
+
+	static final class ThrowOnRCloseResource implements Resource<Object> {
+
+		@Override
+		public Object get() throws Exception {
+			return "foo";
+		}
+
+		@Override
+		public void close() throws Exception {
+			throw EXPECTED_THROW_ON_R_CLOSE_EXCEPTION;
+		}
+
+	}
+
+	private static final Exception EXPECTED_THROW_ON_R_CLOSE_EXCEPTION = new UnknownHostException(
+		"wait, where's the Internet gone?!");
+
+	// ---
+
+	@DisplayName("when a test class has a test method with a parameter annotated with both @New and @Shared")
+	@Nested
+	class WhenTestClassHasTestMethodWithParameterAnnotatedWithBothNewAndSharedTests {
+
+		@DisplayName("then an exception is thrown")
+		@Test
+		void thenExceptionIsThrown() throws Exception {
+			ExecutionResults executionResults = PioneerTestKit
+					.executeTestClass(TestMethodWithParameterAnnotatedWithBothNewAndShared.class);
+			Method failingTest = TestMethodWithParameterAnnotatedWithBothNewAndShared.class
+					.getDeclaredMethod("theTest", String.class);
+			assertThat(executionResults)
+					.hasSingleFailedTest()
+					.withExceptionInstanceOf(ParameterResolutionException.class)
+					.hasMessage("Parameter [%s] in method [%s] is annotated with both @New and @Shared",
+						failingTest.getParameters()[0], failingTest);
+		}
+
+	}
+
+	static class TestMethodWithParameterAnnotatedWithBothNewAndShared {
+
+		@Test
+		void theTest(
+				@New(DummyResourceFactory.class) @Shared(factory = DummyResourceFactory.class, name = "some-name") String param) {
+			fail("We should not get this far.");
+		}
+
+	}
+
+	static final class DummyResourceFactory implements ResourceFactory<String> {
+
+		@Override
+		public Resource<String> create(List<String> arguments) {
+			return () -> "dummy";
+		}
+
+	}
+
+	// ---
+
+	@DisplayName("when a parameter is annotated with @Shared, and another parameter is annotated with @Shared with "
+			+ "the same name but a different factory type")
+	@Nested
+	class WhenParameterIsAnnotatedWithSharedAndAnotherParamIsAnnotatedWithSharedWithSameNameButDifferentFactoryTypeTests {
+
+		@DisplayName("then it throws an exception")
+		@Test
+		void thenItThrowsAnException() {
+			ExecutionResults executionResults = PioneerTestKit
+					.executeTestClass(SingleTestMethodWithConflictingSharedTempDirParametersTestCases.class);
+			executionResults
+					.allEvents()
+					.debug()
+					.assertThatEvents()
+					.haveExactly(//
+						1, //
+						finished(//
+							throwable(//
+								instanceOf(ParameterResolutionException.class), //
+								message(String
+										.format(
+											"Two or more parameters are annotated with @Shared annotations with the "
+													+ "name \"%s\" but with different factory classes",
+											"some-name")))));
+		}
+
+	}
+
+	static class SingleTestMethodWithConflictingSharedTempDirParametersTestCases {
```

Comment:
Shouldn't this also throw an exception?

```java
@Test
void test1(@Shared(factory = DummyResourceFactory.class, name = "some-name") String param) { }

@Test
void test2(@Shared(factory = OtherResourceFactory.class, name = "some-name") String param) { }
```

Reply:
@nipafx Oh, great catch! I'm glad you caught this, as it forced me to write a test that exposed a parallelism bug in the code, which I've fixed. See commit https://github.com/junit-pioneer/junit-pioneer/pull/491/commits/620900bbae5e16e38be643462d6db408cedcafc9. :D